tests/playbooks/tests_wireless.yml

```
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+  vars:
+    interface: wlan0
+  tasks:
+    - name: "INIT: wireless tests"
+      debug:
+        msg: "##################################################"
+    - include_tasks: tasks/setup-mock-wifi.yml
+    - name: Copy client certs
+      copy:
+        src: "{{ item }}"
+        dest: "/etc/pki/tls/{{ item }}"
+        mode: 0644
+      with_items:
+        - client.key
+        - client.pem
+        - cacert.pem
+    - block:
+        - name: "TEST: wireless connection with WPA-PSK"
+          debug:
+            msg: "##################################################"
+        - import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface }}"
+                state: up
+                type: wireless
+                ip:
+                  address:
+                    - 203.0.113.2/24
+                  dhcp4: "no"
+                  auto6: "no"
+                wireless:
+                  ssid: "mock_wifi"
+                  key-mgmt: "wpa-psk"
+                  password: "p@55w0rD"
+        - import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface }}"
+                persistent_state: absent
+                state: absent
```

```suggestion
                state: down
```

The runtime state should be `down`.

---

fixed